modules/docs/lib/widgets/enhancedComponent.tsx

```diff
@@ -0,0 +1,199 @@
+  Border: ['utils/border'],
+  Depth: ['utils/depth'],
+  Text: ['utils/text'],
+  Other: ['utils/other'],
+};
+
+registerWidget<EnhancedComponentValue>('enhancedComponent', ({value, doc, meta}) => {
+  const groups = groupProps(value.props);
+  const parentComponentName = React.useContext(ParentComponentNameContext);
+  const parentComponentJSDoc = React.useContext(ParentComponentJSDocContext);
+  const hideHeader = meta && meta.hideHeader === true;
+
+  const intro = (
+    <>
+      {!parentComponentName && !hideHeader && value.displayName ? (
+        <Heading>{value.displayName}</Heading>
+      ) : null}
+
+      <ParentComponentNameContext.Provider value="">
+        <ParentComponentJSDocContext.Provider value={defaultJSDoc}>
+          <MdxJSToJSX>{parentComponentJSDoc.description || doc?.description || ''}</MdxJSToJSX>
+        </ParentComponentJSDocContext.Provider>
+      </ParentComponentNameContext.Provider>
+    </>
+  );
+
+  // We don't want to re-document a subcomponent that is a container component with a different
+  // model. If we detect another container, only render the intro and a link to the container
+  // component
+  if (parentComponentName && value.componentType === 'container') {
+    return (
+      <>
+        {intro}
+        <MDX as="p">
+          This component references the{' '}
+          <ParentComponentNameContext.Provider value="">
+            <ParentComponentJSDocContext.Provider value={defaultJSDoc}>
+              <SymbolDialog value={{kind: 'symbol', name: value.displayName || ''}} />
+            </ParentComponentJSDocContext.Provider>
+          </ParentComponentNameContext.Provider>{' '}
+          component.
+        </MDX>
+      </>
+    );
+  }
+
+  return (
+    <>
+      {intro}
+      {value.styleComponent ? (
+        <>
+          <Heading headingOffset={1}>Style Component</Heading>
+          <MDX as="p">
+            <code>
+              <ParentComponentJSDocContext.Provider value={defaultJSDoc}>
+                <Value value={value.styleComponent} />
+              </ParentComponentJSDocContext.Provider>
+            </code>
+          </MDX>
+        </>
+      ) : null}
+      <Heading headingOffset={1}>Props</Heading>
+      {value.baseElement && (
+        <MDX as="p">
+          Props extend from <Value value={value.baseElement} />. Changing the <code>as</code> prop
+          will change the element interface.
+        </MDX>
+      )}
+      {value.componentType === 'container' && value.model ? (
+        <MDX as="p">
+          Props extend from <SymbolDialog value={{kind: 'symbol', name: `${value.model}Config`}} />.
+          If a <code>model</code> is passed, model config is ignored.
+        </MDX>
+      ) : null}
+      {Object.keys(groups).map(key => {
+        return (
+          <React.Fragment key={key}>
+            {key === 'Local' ? (
+              <>
+                <PropertiesTable properties={groups[key]}></PropertiesTable>
```

Comment:
and not a big deal but this could be different (stands out in PR review because it wraps to another line):
```suggestion
                <PropertiesTable properties={groups[key]} />
```